templates/terraform/nested_query.go.erb

```diff
@@ -13,7 +13,9 @@ func flattenNested<%= resource_name -%>(d *schema.ResourceData, meta interface{}
   <% end -%>
   v, ok = res["<%=object.nested_query.keys[-1]-%>"]
   if !ok || v == nil {
-    return nil, nil
+    // It's possible that there is only one of these resources and
+    // that res represents that resource.
+    v = res
```

Comment:
I'm not sure I follow- if key `foos` doesn't exist, we assume the object is a `foo`? Does the API promote a single element instead of returning it in a list?

Reply:
Yep, this one does.  :(

Reply:
Have you filed a bug against the API? That's, uh, traitful.

Reply:
Not from their perspective.  It's only so odd because it's different convention than other list-only APIs.  Specifically, this API returns a single element from the operation, but a list from their list API.  The other list-only APIs return a list in both places, so we use this method in both places.  This one is _technically_ out of compliance with the "should"s in https://aip.dev/151, but it's also out of compliance with the "should" in https://aip.dev/131, and the out-of-compliances sort of balance each other out.

Reply:
Ah, got it. You were already talking to them about adding GET, right?

Reply:
Yep!